### PR TITLE
Add sensible defaults to the exclude list

### LINKED ISSUE #52: Work with govuk to see how they are using detect-secrets, and if it is compatible with our hook
bug fixing to allow GOV.UK devs to run the command successfully

### LINKED ISSUE #71: Add package-lock, python pip files and other common package files to the exclude field in the yaml
The global install yaml file has an exclude option that we could add common package files to, in order to save people a bunch of time. We have had more than one request come in about how to auto exclude a package file due to the number of high entropy strings.

https://github.com/alphagov/gds-pre-commit/blob/master/global_install/hooks/.pre-commit-config.yaml#L13

One example could be (not tested, just read the docs quickly):

```
    hooks:
    -   id: detect-secrets
        args: ['--baseline', '.secrets.baseline']
        exclude: |
              (?x)(
                     ^package-lock.json$/|
                     ^Pipfile/
              )
```

### PR DESCRIPTION
**Pull request by 0atman**
such as package locking files.
Fixes https://github.com/alphagov/gds-pre-commit/issues/71

### CODE REVIEW

File: global_install/hooks/.pre-commit-config.yaml [hunk 1]
```diff
@@ -10,4 +10,9 @@ repos:
     hooks:
     -   id: detect-secrets
         args: ['--baseline', '.secrets.baseline']
-        exclude: .*/tests/.*
+        exclude: .*/tests/.*|
+              (?x)(
+                     ^package-lock.json$/|
+                     ^Pipfile/|
+                     ^pyproject.toml
```
Comment (0atman) [inline global_install/hooks/.pre-commit-config.yaml:17]: Any more common hash-y files you can think of?

File: global_install/hooks/.pre-commit-config.yaml [hunk 2]
```diff
@@ -10,4 +10,9 @@ repos:
     hooks:
     -   id: detect-secrets
         args: ['--baseline', '.secrets.baseline']
-        exclude: .*/tests/.*
+        exclude: .*/tests/.*|
```
Comment (akinnane) [inline global_install/hooks/.pre-commit-config.yaml]: I don't think we should blanket exclude tests. Test data is a prima the facia case where real secrets might be committed.

File: global_install/hooks/.pre-commit-config.yaml (hunk 1 above)
Comment (akinnane) [reply global_install/hooks/.pre-commit-config.yaml:17]: I think this should be added to a step in the documentation so users explicitly know which files are, and are not scanned for committed secrets. While this helps adoption it adds a non obvious behaviour.

File: global_install/hooks/.pre-commit-config.yaml (hunk 2 above)
Comment (0atman) [reply global_install/hooks/.pre-commit-config.yaml]: That's a really good point. Nice catch!

File: global_install/hooks/.pre-commit-config.yaml (hunk 1 above)
Comment (0atman) [reply global_install/hooks/.pre-commit-config.yaml:17]: Already there, if you remember, you razzed me for my `cat` use :laughing: 
https://github.com/alphagov/gds-pre-commit#ignoring-secrets

Comment (0atman) [reply global_install/hooks/.pre-commit-config.yaml:17]: I REMOVED THE :cat:

Comment (akinnane) [reply global_install/hooks/.pre-commit-config.yaml:17]: Should these additional ignore be added to the documentation too then?

Comment (0atman) [reply global_install/hooks/.pre-commit-config.yaml:17]: Yeah, good point. It's actually in the default example config, which I have PR'd. https://github.com/Yelp/detect-secrets/pull/340

**Comment by 0atman** [comment]
This is now ready

**Review by danjoneslf — APPROVED** [review]
Gemfile.lock doesn't seem to have anything high entropy. I don't know if there might be equivalents in composer, bower, yarn, webpack or if we use them but I guess it's no hardship to add more if people find them.

**Comment by 0atman** [comment]
Oh yeah, so it doesn't. Just version numbers, my mistake.

**Pull request merged**